docs/source/cpp/compute.rst

```diff
@@ -0,0 +1,419 @@
+| isin                     | Unary      | Boolean,Numeric, Temporal        | Boolean (2)           | :struct:`SetLookupOptions`                 |
++--------------------------+------------+----------------------------------+-----------------------+--------------------------------------------+
+| match                    | Unary      | Binary- and String-like          | Int32 (3)             | :struct:`SetLookupOptions`                 |
++--------------------------+------------+----------------------------------+-----------------------+--------------------------------------------+
+| match                    | Unary      | Null                             | Int32 (3)             | :struct:`SetLookupOptions`                 |
++--------------------------+------------+----------------------------------+-----------------------+--------------------------------------------+
+| match                    | Unary      | Boolean,Numeric, Temporal        | Int32 (3)             | :struct:`SetLookupOptions`                 |
++--------------------------+------------+----------------------------------+-----------------------+--------------------------------------------+
+
+* \(1) Output is true iff :member:`BinaryContainsExactOptions::pattern`
+  is a substring of the corresponding input element.
+
+* \(2) Output is true iff the corresponding input element is equal to one
+  of the elements in :member:`SetLookupOptions::value_set`.
+
+* \(3) Output is the index of the corresponding input element in
+  :member:`SetLookupOptions::value_set`, if found there.  Otherwise,
+  output is null.
+
+Structural transforms
+~~~~~~~~~~~~~~~~~~~~~
+
++--------------------------+------------+--------------------+---------------------+---------+
+| Function name            | Arity      | Input types        | Output type         | Notes   |
++==========================+============+====================+=====================+=========+
+| is_null                  | Unary      | Any                | Boolean             | \(1)    |
++--------------------------+------------+--------------------+---------------------+---------+
+| is_valid                 | Unary      | Any                | Boolean             | \(2)    |
++--------------------------+------------+--------------------+---------------------+---------+
+| list_value_lengths       | Unary      | List-like          | Int32 or Int64      | \(3)    |
++--------------------------+------------+--------------------+---------------------+---------+
+
+* \(1) Output is true iff the corresponding input element is non-null.
+
+* \(2) Output is true iff the corresponding input element is null.
+
+* \(3) Each output element is the length of the corresponding input element
+  (null if input is null).
+
+Conversions
+~~~~~~~~~~~
+
+A general conversion function named ``cast`` is provided which accepts a large
+number of input and output types.  The type to cast to can be passed in a
+:struct:`CastOptions` instance.
```

Comment:
How?

Reply:
You get the answer by clicking on the `CastOptions` link not rendered :-) I'm not sure it's worth repeating here.